ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogWorker.java

```diff
@@ -356,7 +361,9 @@ private boolean shouldFlush() {
     } else if (pendingFlushNum >= forceSyncNum) {
       return true;
     }
-    return pendingFlushNum > 0 && queue.isEmpty();
+
+    final Task nextTask = queue.peek();
+    return pendingFlushNum > 0 && (nextTask == null || nextTask.getType() != TaskType.WRITE_LOG);
```

Comment:
Let's simply use `instanceof`.
```java
    final Task peeked = queue.peek();
    // when the queue is empty, peek() returns null and (null instanceof WriteLog) is false
    return pendingFlushNum > 0 && !(peeked instanceof WriteLog);
```

Reply:
Thank for the review, @szetszwo . I've simplified it to use instanceof now. Please review again.